pkg/trace/agent/agent.go

```diff
@@ -255,12 +250,12 @@ func (a *Agent) Process(p *api.Payload, sublayerCalculator *stats.SublayerCalcul
 			ss.Size += pb.Trace(events).Msgsize()
 		}
 		if ss.Size > writer.MaxPayloadSize {
-			a.Out <- ss
+			a.TraceWriter.In <- ss
```

Comment:
One concern with this approach is that someone external could be reading from the channel, messing up the flow. A solution to that would be to make the channel unexported and have a method like:
```go
func (*TraceWriter) In() chan<- *SampledSpans
```
Then, the returned channel can only be sent to...

Happy to make this change.